src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -90,6 +90,16 @@ const DomainFragment = ({
 
   const isReady = useIsReady();
 
+  React.useEffect(() => {
+    if (searchText.length === 0) {
+      setActiveSections([]);
+    } else if (searchText.length >= 0 && followedDomains.length > 0) {
+      setActiveSections([0]);
+    } else {
+      setActiveSections([]);
+    }
+  }, [searchText, followedDomains]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in the `useEffect` hook seems to be incorrect. The condition `searchText.length >= 0 && followedDomains.length > 0` will always be true when `searchText.length === 0` is false, because `searchText.length` cannot be less than 0. This means that the last else block will never be executed. If you want to set active sections to [0] only when there are followed domains and search text, you should check if `searchText.length > 0` instead of `searchText.length >= 0`.

```diff
-    } else if (searchText.length >= 0 && followedDomains.length > 0) {
+    } else if (searchText.length > 0 && followedDomains.length > 0) {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->